src/mon/OSDMonitor.cc

```diff
@@ -431,10 +431,12 @@ void OSDMonitor::on_active()
     mon->clog->info() << "osdmap " << osdmap << "\n";
 
   if (!mon->is_leader()) {
-    list<MOSDFailure*> ls;
+    list<MonOpRequestRef> ls;
     take_all_failures(ls);
     while (!ls.empty()) {
-      dispatch(ls.front());
+      MonOpRequestRef op = ls.front();
+      op->mark_osdmon_event(__func__);
+      dispatch(op);
```

Comment:
the only place where we reset the `MonOpRequestRef` is when we remove the failure messages at seeing an "alive" `MOSDFailure` message from a certain OSD. in that case, if the monitor is not active after the reports from that OSD are removed, and re-dispatches the empty `MonOpRequestRef` at https://github.com/ceph/ceph/pull/5021/files#diff-0a5db46a44ae9900e226289a810f10e8R439 .

then `OSDMonitor::dispatch()` will dereference a NULL `PaxosServiceMessage` at https://github.com/ceph/ceph/pull/5021/files#diff-0b9efd13545ddf68b5de9c579f78cb78R39 . bang!

so, i'd suggest we check these `op`s for NULL before marking and dispatching them.